python/src/context.rs

```diff
@@ -97,6 +99,34 @@ impl ExecutionContext {
         Ok(())
     }
 
+    #[args(
+        has_header = "true",
+        delimiter = "b\",\"",
+        schema_infer_max_records = "1000",
+        file_extension = "\".csv\""
+    )]
+    fn register_csv(
+        &mut self,
+        name: &str,
+        path: &str,
+        has_header: bool,
+        delimiter: &[u8],
```

Comment:
Here we should have a `Schema` argument exposed as well, but I noticed that FFI hasn't been implemented for `Schema` and `DataType` objects in `arrow-rs`. We should probably expose all of the `ArrowSchema` based structs there first, then convert pyarrow objects using the C interface rather than calling out to python functions (like the datatype python bindings are currently implemented).

Reply:
I agree. I only learnt that the Schema and DataType have a c data interface recently. This likely requires some refactoring on the arrow-rs, as it assumes that metadata do not require a specific in-memory alignment, and yet the c data interface makes such requirement.